client/src/push-content.ts

```diff
@@ -32,7 +33,34 @@ export const getRepo = (): Repository => {
   return result
 }
 
+export const taggingDialog = async (): Promise<string | undefined> => {
+  /* instanbul ignore next */
+  return await vscode.window.showInformationMessage(
+    'Are you tagging for release?',
+    // { modal: true },
+    ...['No Tag', 'Release Candidate', 'Release']
+  )
+}
+
+export const getNewTag = (repo: Repository, release: boolean): string | undefined => {
+  /* instanbul ignore next */
+  const tags: integer[] = []
+
+  // could probaly use a filter-reduce here
+  repo.state.refs.forEach(ref => {
+    if (ref.type !== RefType.Tag) { return }
+    if (release === ref.name?.includes('rc')) { return }
+
+    const versionNumberString = expect(ref.name, '').replace('rc', '')
+    tags.push(Number(versionNumberString))
+  })
+
+  const previousVersion = tags.length > 0 ? tags[tags.length - 1] : 0
```

Comment:
Yes, the ref's dont have any timestamps or other ordering information I could find, and, it does appear to be in order.

Reply:
Right, but it's at least possible for the tags to be out of order, correct? Since there's an underlying assumption here that these tags will be numbers anyways, would it be safer to take the largest number in the array instead of the last entry just in case?

Reply:
makes sense